home/service/search.py

```diff
@@ -76,6 +76,42 @@ def _get_context(self) -> dict[str, Any]:
             "paginator": self.paginator,
             "total_results": self.results.total_results,
             "label_clear_href": label_clear_href,
+            "readable_match_reasons": self._get_match_reason_display_names(),
         }
 
         return context
+
+    def _highlight_results(self):
+        "Take a SearchResponse and add bold markdown where the query appears"
+        string_to_highlight = self.form.cleaned_data.get("query") if self.form.is_valid() else ""
+        highlighted_results = deepcopy(self.results)
+
+        if string_to_highlight == "":
```

Comment:
Will the `self.form.cleaned_data.get("query")` return `"*"` if there's no query (as in native DataHub)? If so, should we handle it here?
```suggestion
        if string_to_highlight in ("", "*"):
```

Reply:
The "*" is set as the default query string if nothing is passed to the catalogue API library, so it'll still be a blank string in the form side afaik?

https://github.com/ministryofjustice/data-platform/blob/2622d7cbde6a34d34dfaf993088795b616f6e5da/python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py#L45

Reply:
I think we can use either interchangably to search for everything

Reply:
I'm just gonna add it, I don't think there's any harm